src/pages/SubmissionDetails.js

```diff
@@ -42,6 +43,24 @@ export const SubmissionDetails = ({ submissionId }) => {
       Utils.differenceFromNowInSeconds(submitted)
   }
 
+  const getFilter = filter => {
+    let filterStatement
+    switch (filter) {
+      case 'Error':
+        filterStatement = _.filter(r => errorStates.includes(r.state), runsData)
```

Comment:
I went back and forth on whether or not this is too much of a nit-pick, but in the end decided it was worth saying...

in each of these cases, you're not just defining a filter - you're actually performing the filter operation. So `getFilter` isn't quite accurate - it's really something like `getFilteredRunsData` (or something)

You could just rename the function, which would be fine. Or...

if you wanted, you could make some small adjustments to this function to make it more "functional" (as in "functional programming"): just drop the `runsData` in this line, and line 56.

Why does that work? because in `lodash/fp`, functions that are called without all of its arguments automatically return another function that will accept the missing arguments (this is called "currying").

if you do this, then this function will truly be returning a "filter" -- a function that accepts `runsData` and returns filtered data.

(lmk if you want to discuss this in more detail)

Reply:
* (why is all of the above a good thing? because if you get rid of `runsData`, the function becomes "pure" (i.e. it has no side effects), which is generally considered to be easier to reason about and test